Error: Cannot find module 'p-limit'

<!-- DO NOT DELETE
validate_template=true
template_path=.github/ISSUE_TEMPLATE/bug_report.md
-->

<!--
Thank you for contributing to the Firebase community!

Think you found a bug?
=======================
Yeah, we're definitely not perfect! Please use this template and include a minimal repro when opening the issue. If you know how to solve the issue, please create a Pull Request, and we'd be happy to review it!

Have a feature request?
========================
Great, we love hearing how we can improve our products! However, do not use this template to submit a feature request. Please submit your feature requests to: https://firebase.google.com/support/contact/bugs-features/

Have a usage question?
=======================
We get lots of those and we love helping you, but GitHub is not the best place for them and they will be closed. Please take a look at the guide first: https://firebase.google.com/docs/cli/

If the official documentation doesn't help, try asking through our official support channel: https://firebase.google.com/support/

Additional locations to check for solutions or assistance from the community:
- Stack Overflow: https://stackoverflow.com/
- Firebase Slack Community: https://firebase.community/

*Please avoid duplicate posting across multiple channels!*
-->

### [REQUIRED] Environment info

<!-- What version of the Firebase CLI (`firebase-tools`) are you using? Note that your issue may already be fixed in the latest versions. The latest version can be found at https://github.com/firebase/firebase-tools/releases -->

<!-- Output of `firebase --version` -->

**firebase-tools:** v11.25.0

<!-- e.g. macOS, Windows, Ubuntu -->

**Platform:** macOS

### [REQUIRED] Test case

<!-- Provide a minimal, complete, and verifiable example (http://stackoverflow.com/help/mcve) -->

see below

### [REQUIRED] Steps to reproduce

<!-- Provide the steps needed to reproduce the issue with the above test case. -->

```bash
nvm i 18 
npm i -g firebase-tools
firebase login
```

Turned out I _was_ logged in - downgrading to v11.24 worked fine.

### [REQUIRED] Expected behavior

<!-- What is the expected behavior? -->

Not a crash!

### [REQUIRED] Actual behavior

<!-- Run the command with --debug flag, and include the logs below. -->

```
» firebase login
node:internal/modules/cjs/loader:1078
  throw err;
  ^

Error: Cannot find module 'p-limit'
Require stack:
- /Users/bkend/.nvm/versions/node/v18.15.0/lib/node_modules/firebase-tools/lib/database/import.js
- /Users/bkend/.nvm/versions/node/v18.15.0/lib/node_modules/firebase-tools/lib/commands/database-import.js
- /Users/bkend/.nvm/versions/node/v18.15.0/lib/node_modules/firebase-tools/lib/commands/index.js
- /Users/bkend/.nvm/versions/node/v18.15.0/lib/node_modules/firebase-tools/lib/index.js
- /Users/bkend/.nvm/versions/node/v18.15.0/lib/node_modules/firebase-tools/lib/bin/firebase.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
    at Module._load (node:internal/modules/cjs/loader:920:27)
    at Module.require (node:internal/modules/cjs/loader:1141:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at Object.<anonymous> (/Users/bkend/.nvm/versions/node/v18.15.0/lib/node_modules/firebase-tools/lib/database/import.js:8:19)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1308:10)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)
    at Module.require (node:internal/modules/cjs/loader:1141:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at Object.<anonymous> (/Users/bkend/.nvm/versions/node/v18.15.0/lib/node_modules/firebase-tools/lib/commands/database-import.js:8:18)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1308:10)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/bkend/.nvm/versions/node/v18.15.0/lib/node_modules/firebase-tools/lib/database/import.js',
    '/Users/bkend/.nvm/versions/node/v18.15.0/lib/node_modules/firebase-tools/lib/commands/database-import.js',
    '/Users/bkend/.nvm/versions/node/v18.15.0/lib/node_modules/firebase-tools/lib/commands/index.js',
    '/Users/bkend/.nvm/versions/node/v18.15.0/lib/node_modules/firebase-tools/lib/index.js',
    '/Users/bkend/.nvm/versions/node/v18.15.0/lib/node_modules/firebase-tools/lib/bin/firebase.js'
  ]
}

Node.js v18.15.0
```

Yes. It broke our GitHub Actions.
```
Error: Cannot find module 'p-limit'
Require stack:
- /home/runner/.config/yarn/global/node_modules/firebase-tools/lib/database/import.js
- /home/runner/.config/yarn/global/node_modules/firebase-tools/lib/commands/database-import.js
- /home/runner/.config/yarn/global/node_modules/firebase-tools/lib/commands/index.js
- /home/runner/.config/yarn/global/node_modules/firebase-tools/lib/index.js
- /home/runner/.config/yarn/global/node_modules/firebase-tools/lib/bin/firebase.js
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,0 +1,1 @@
+- Fixes a missing dependency on `p-limit`. (#5619)

--- a/package.json
+++ b/package.json
@@ -132,6 +132,7 @@
     "node-fetch": "^2.6.7",
     "open": "^6.3.0",
     "ora": "^5.4.1",
+    "p-limit": "^3.0.1",
     "portfinder": "^1.0.32",
     "progress": "^2.0.3",
     "proxy-agent": "^5.0.0",

--- a/src/database/import.ts
+++ b/src/database/import.ts
@@ -7,7 +7,7 @@ import * as StreamObject from "stream-json/streamers/StreamObject";
 import { URL } from "url";
 import { Client, ClientResponse } from "../apiv2";
 import { FirebaseError } from "../error";
-import * as pLimit from 'p-limit';
+import * as pLimit from "p-limit";
 
 const MAX_CHUNK_SIZE = 1024 * 1024 * 10;
 const CONCURRENCY_LIMIT = 5;

--- a/src/database/import.ts
+++ b/src/database/import.ts
@@ -4,11 +4,10 @@ import * as Filter from "stream-json/filters/Filter";
 import * as stream from "stream";
 import * as StreamObject from "stream-json/streamers/StreamObject";
 
-import pLimit from "p-limit";
-
 import { URL } from "url";
 import { Client, ClientResponse } from "../apiv2";
 import { FirebaseError } from "../error";
+import * as pLimit from 'p-limit';
 
 const MAX_CHUNK_SIZE = 1024 * 1024 * 10;
 const CONCURRENCY_LIMIT = 5;

